File: problem2/problem2.py
def problem2(n):
    sum = 0
    f1 = f2 = fn = 1
    while f2 < n:
        if fn % 2 == 0:
            sum += fn 
        fn = f1 + f2
        f1, f2 = f2, fn
    return sum

if __name__ == '__main__':
    print(problem2(4000000))

File: problem2/problem2.hs
-- from rosettacode.org
fib = 0 : 1 : zipWith (+) fib (tail fib)

problem2 n =
  sum (takeWhile (< n) [x | i <- [1..], let x = (fib !! i), even x])

main :: IO()
main =
  putStrLn (show (problem2 4000000))
